ws-conn/ws-connection-handler.go

```diff
@@ -109,16 +123,24 @@ func (handler *ConnectionHandlerImpl) readDataFromConnections(inputChan chan []b
 
 // Send ping message to the connection client.
 func (*ConnectionHandlerImpl) sendPingMessage(wsConn *websocket.Conn) {
-	ticker := time.NewTicker(PingPeriod)
+	ticker := time.NewTicker(pingPeriod)
 	defer ticker.Stop()
 
 	for range ticker.C {
-		err := wsConn.WriteMessage(websocket.PingMessage, []byte{})
-		if err != nil {
-			if !IsNormalWSError(err) {
-				logrus.Errorf("Error occurs on sending ping message to ws-conn. %v", err)
-			}
-			return
+		if err := wsConn.WriteMessage(websocket.PingMessage, []byte{}); err != nil {
+			break
+		}
+	}
+}
+
+func (handler *ConnectionHandlerImpl) CloseConnections() {
+	defer handler.wsConnsLock.Unlock()
+	handler.wsConnsLock.Lock()
+
+	for index, wsConn := range handler.wsConns {
+		if err := wsConn.Close(); err != nil {
+			logrus.Errorf("failed to close connection. Cause: %v", err)
 		}
+		handler.wsConns = append(handler.wsConns[:index], handler.wsConns[index+1:]...)
```

Comment:
Are you sure that it's going to work as expected?

```go
type TestArrayHandler struct {
	array     []int
}

func main() {
	test := TestArrayHandler{
		array: []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
	}
	for index, value := range test.array {
		logrus.Info(value)
		test.array = append(test.array[:index], test.array[index+1:]...)
	}
}
```
=> 
```
INFO[0000] 1                                            
INFO[0000] 3                                            
INFO[0000] 5                                            
INFO[0000] 7                                            
INFO[0000] 9                                            
INFO[0000] 10                                           
panic: runtime error: slice bounds out of range [6:5]

```
Maybe we should just assign empty slice value after loop?

Reply:
Yes, you are right, thanks. I tested code with one client(one array element). That why I didn't see this an error.